src/core/net/dns_headers.hpp

```diff
@@ -735,6 +735,19 @@ class Name
      */
     static otError CompareName(const Message &aMessage, uint16_t &aOffset, const Message &aMessage2, uint16_t aOffset2);
 
+    /**
+     * This static method tests if a DNS name ends with given domain.
+     *
+     * @p aName and @p aDomain MUST (or not) end with dot ('.') at the same time.
```

Comment:
I think we can disallow strings that do not end with dot unless it's actually useful.

Reply:
I think when we read name from `message` we always append an ending  `.` to it (which if I recal correctly is the recommend way to represent the DNS names).

We can also consider here to handle it in the implementation of this method and allow either name to end or not end with `.` (basically making this method handle all cases, and making it more flexible to use).

Thoughts?

Reply:
I would prefer to make it more strict in the OT core logic so as to make the code simpler (handling less conditions). 

For example, if we allow domain names not to end with dot, then one question is that should we allow a empty domain (i.e. "") ?

Also, if we allow the domain names not to end with dot, then this method should work no matter `aName` or `aDomain` ends with dot. For example, it should work when `aName` ends with dot but `aDomain` does not. 

Thoughts? @wgtdkp @abtink

Reply:
There are no server side cases that the names will end without `.`, but I think there could such cases on the client side (e.g. the name or domain given by the user). I will change the implementation to allow it.